### PR TITLE
add support for init step (all steps except packer)

### PR DESCRIPTION
**Pull request by simonrondelez**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,18 @@ endif
...
+	test -n "$(FUNCTION)"  # $$FUNCTION
+	test -n "$(PUPPET_REPO)"  # $$PUPPET_REPO
+	test -n "$(PACKER_PROFILE)" #$$PACKER_PROFILE
+
+	if [[ -d "puppet/" ]] && ! git --git-dir=./puppet/.git remote get-url origin | grep --quiet $(PUPPET_REPO); then rm -rf ./puppet; fi
+	if [[ ! -d "puppet/" ]]; then git clone -b $(GIT_BRANCH) $(PUPPET_REPO) puppet; fi
+	cd puppet && git pull && git checkout $(GIT_BRANCH) && git submodule update --init --recursive
+	if [[ -e "puppet/Puppetfile" ]]; then cd puppet; r10k puppetfile install; fi
 
 build:
```
Comment (duboisph) [inline Makefile]: You can remove the duplicate code and just do:
`build: init`

**Review by duboisph — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (iuriaranda) [inline Makefile]: you can run the `init` task as a pre-requisite of `build`, so you don't repeat yourself.

```
build: init
...
```

this should do the trick

Comment (duboisph) [reply Makefile]: I was first :P

Comment (iuriaranda) [reply Makefile]: lol

**Review by iuriaranda — APPROVED** [review]

**Review by duboisph — APPROVED** [review]

**Pull request merged**